visualizer/draw/bubbleprof-ui.js

```diff
@@ -358,43 +358,39 @@ class BubbleprofUI extends EventEmitter {
   }
 
   generateCollapsedNodeHash (uiWithinCollapsedNode) {
-    let hash = `l${uiWithinCollapsedNode.layoutNode.id}|`
+    let hashString = uiWithinCollapsedNode.layoutNode.id
     const appendParentNode = (parentUI) => {
-      if (!parentUI.layoutNode) {
-        hash += 'm'
-      } else {
+      if (parentUI.layoutNode) {
         const dataNode = parentUI.layoutNode.node
         switch (dataNode.constructor.name) {
           case 'ClusterNode':
-            hash += 'c' + dataNode.clusterId
+            hashString += '-c' + dataNode.clusterId
             break
           case 'ArtificialNode':
-            hash += `l${parentUI.layoutNode.id}|`
+            hashString += `-${parentUI.layoutNode.id}`
             appendParentNode(parentUI.parentUI)
             break
         }
       }
     }
     appendParentNode(uiWithinCollapsedNode.parentUI)
-    return hash
+    return hashString
```

Comment:
Currently this generates hierarchy in right-to-left order which I think is counterintuitive
It would be a `nice to have` to flip the order to left-to-right
i.e.
When storing a clump inside a cluster (`c9 -> x1`) in url we get
`#x1-c9`, but I would argue `#c9-x1` is more human parseable
(for making dev lives' easier, the end user doesn't care anyway)

Reply:
Not sure, the idea is the first id after the hash is the selected one, so the first character of the hash string always tells you what type it is.

 If we reverse the order people would probably incorrectly assume the first id is in the main layout, second is one more deep, etc.

It's like x2-x3-c3 means "select x2, which you'll find in x3, which you'll find in c3"

Reply:
To me this:
`select x2, which you'll find in x3, which you'll find in c3`
is the same as this:
`find c3, then inside it find x3, then inside it find x2`
Personally I find second more natural to follow, but that is subjective

> If we reverse the order people would probably incorrectly assume the first id is in the main layout

If we take above and consider scenario where c3 isn't in Main View, then it must be inside a clump. Therefore both paths would have additional `x` entry, e.g. `x1`:
`select x2, which you'll find in x3, which you'll find in c3, which you'll find in x1`
or
`find x1, then inside it find c3, then inside it find x3, then inside it find x2`

>  If we reverse the order people would probably incorrectly assume [...] second is one more deep

Tbh I had the same assumption with current setup until you used the word `jump`, then I remembered

Anyway it's not a big deal to me ¯\\\_(ツ)\_/¯